### PR TITLE
Integrate JavaScript linting solution

### PR DESCRIPTION
**Pull request by chris-kobrzak**
Issue #5

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -33,6 +33,9 @@ module.exports = function(config) {
 
     webpack: {
       module: {
+       preLoaders: [
+         { test: /\.jsx?$/, loader: "eslint", exclude: /(node_modules)/ }
```
Comment (chris-kobrzak) [inline karma.conf.js]: Note to self: explain why we want to use preloaders as we might as well put the ESLint loader in the loaders section below.

**Comment by chris-kobrzak** [comment]
Tidied up and squashed commits, addressed points raised in @jacobwalton's code review so this PR is finally ready.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    space-before-function-paren: [2, "never"]
+    space-infix-ops: 2
+    space-return-throw-case: 2
+    space-unary-ops: [2, {words: true, nonwords: true}]
+    spaced-comment: [2, "always", { exceptions: ["-"]}]
+    strict: 0
+    use-isnan: 2
+    valid-jsdoc: [2, { prefer: { "return": "returns"}}]
+    wrap-iife: 2
+    yoda: [2, "never"]
```
Comment (jacobwalton) [inline .eslintrc]: There are a few more recommended items under [Possible Errors](http://eslint.org/docs/rules/#possible-errors) in the docs that we should probably add, specifically:
* [`no-dupe-args`](http://eslint.org/docs/rules/no-dupe-args)
* [`no-extra-semi`](http://eslint.org/docs/rules/no-extra-semi)
* [`no-invalid-regexp`](http://eslint.org/docs/rules/no-invalid-regexp)
* [`valid-typeof`](http://eslint.org/docs/rules/valid-typeof)

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    no-ex-assign: 2
+    no-extend-native: 2
+    no-extra-bind: 2
+    no-extra-boolean-cast: 2
+    no-extra-parens: 0
+    no-fallthrough: 2
+    no-floating-decimal: 2
+    no-func-assign: 2
+    no-implied-eval: 2
+    no-inner-declarations: 2
```
Comment (jacobwalton) [inline .eslintrc]: We could set this to `[2, "both"]` to cover variable declarations too.

**Comment by chris-kobrzak** [comment]
https://github.com/whichdigital/webpack-react-workflow/issues/5 has now been merged so closing this PR.

**Pull request merged**